metaci/api/views/testmethod_perf.py

```diff
@@ -0,0 +1,284 @@
+import copy
+
+from django.db.models import FloatField
+from django.db.models.functions import Cast
+
+from django.db.models import F, Avg, Count, Q, Min, Max
+
+import django_filters.rest_framework
+from django_filters.widgets import DateRangeWidget
+from django_filters.rest_framework import DateRangeFilter
+
+from rest_framework import generics, exceptions
+
+from metaci.testresults.models import TestResult
+from metaci.build.models import BuildFlow
+from metaci.api.serializers.testmethod_perf import TestMethodPerfSerializer
+from metaci.repository.models import Repository, Branch
+from metaci.plan.models import Plan
+
+from django.db import connection
+
+
+def set_timeout(timeout):
+    """Restrict extremely long Postgres queries. 
+    Theoretically this should reset on the next request but that
+    could use some additional testing"""
+    with connection.cursor() as cursor:
+        cursor.execute("SET LOCAL statement_timeout=%s", [timeout * 1000])
+
+
+class TurnFilterSetOffByDefaultBase(django_filters.rest_framework.FilterSet):
+    """A bit of a hack to allow two filtersets to work together in generating the
+       djago-filter config form but to have only one of them actually filter the
+       output queryset. The other filters a queryset of a sub-select."""
+
+    really_filter = False
+
+    def __init__(self, *args, really_filter=really_filter, **kwargs):
+        self.really_filter = really_filter
```

Comment:
`really_filter` is only used in this method; is there a reason to store it as an instance variable?

Reply:
Another one of those things that made sense at one time but doesn't anymore.